internal/cmdutil/util.go

```diff
@@ -73,3 +77,12 @@ type NoResourceGroupCRDError struct{}
 func (*NoResourceGroupCRDError) Error() string {
 	return "type ResourceGroup not found"
 }
+
+// ResourceGroupCRDNotLatestError is an error type that will be used when a
+// cluster has a ResourceGroup CRD that doesn't match the
+// latest declaration.
+type ResourceGroupCRDNotLatestError struct{}
+
+func (*ResourceGroupCRDNotLatestError) Error() string {
+	return "type ResourceGroup CRD needs update"
```

Comment:
What is the UX here? Once users update to the latest version of kpt, they will get this message on the next attempt to apply. But what happens then? Do we provide an easy was to update? We should make the error message here more actionable.

Reply:
They could run `kpt live install-resource-group`

Reply:
The change updates the pre-steps of apply. If the ResourceGroup CR is installed, but it doesn't match the latest one in the kpt release, then it updates the ResourceGroup CRD in the cluster.
This update fails in the case that users don't have the permission to GET/UPDATE a CRD resource.
So we can change the error message here to be "The ResourceGroup CRD needs update, please run `kpt live install-resource-group`" to update it.

Reply:
Yeah, with the updated error message, this looks good.